include/crm/common/xml.h

```diff
@@ -395,4 +395,90 @@ void crm_xml_sanitize_id(char *id);
 void crm_xml_set_id(xmlNode *xml, const char *format, ...)
     __attribute__ ((__format__ (__printf__, 2, 3)));
 
+
+/*!
+ * \brief Base for directing lib{xml2,xslt} log into standard libqb backend
+ *
+ * This macro implements the core of what can be needed for directing
+ * libxml2 or libxslt error messaging into standard, preconfigured
+ * libqb-backed log stream.
+ *
+ * It's a bit unfortunate that libxml2 (and more sparsely, also libxslt)
+ * emits a single message by chunks (location is emitted separatedly from
+ * the * message itself), so we have to take the effort to combine these
+ * chunks back to single message.  Whether to do this or not is driven
+ * with \p dechunk toggle.
+ *
+ * The form of a macro was chosen for implicit deriving of __FILE__, etc.
+ * and also because static dechunking buffer should be differentiated per
+ * library (here we assume different functions referring to this macro
+ * will not ever be using both at once), preferably also per-library
+ * context of use to avoid clashes altogether.
```

Comment:
It was discussed that a viable alternative, also to avoid the mentioned
deficiencies of being intrusive in the public API space, would be to
place the macro properly to a private header, as the macro is for
intended for internal only use (by design).